### DISCUSSION TITLE
Welcome to xplr Discussions!

### DISCUSSION THREAD

**Discussion by sayanarijit** [sayanarijit/xplr#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mwaitzman** [comment]
When will there be binaries for Windows 10?

**Reply by sayanarijit** [reply]
Hi there, unfortunately I don't have the resources to build and test `xplr` on windows natively,
However, as per [this comment](https://www.reddit.com/r/commandline/comments/mjot3a/xplr_a_hackable_minimal_fast_tui_file_explorer/gtbtvn7), `xplr` should work fine on Windows-Linux subsystem, i.e. you can use the Linux binary on WSL.

**Reply by mwaitzman** [reply]
ah okay thanks

**Comment by caviv** [comment]
Very nice. But are you familiar with Midnight Commander: https://midnight-commander.org/ ? which looks that have everything you have done and more.

**Reply by sayanarijit** [reply]
Hey, thanks. I have heard of `mc` many times but haven't used it. I have even [listed it as an alternative](https://github.com/sayanarijit/xplr/wiki/Alternatives). `xplr` is supposed to be an extension rather than competitor to the existing tools. For e.g. while exploring files using `xplr`, you can drop into `nnn` when you need more speed, `mc` when you need more control, `broot` when you need more visibility etc.

**Comment by lmburns** [comment]
Is there a way to specify a configuration file directly? Or is there a way to make sure that mine is actually being sourced. I've installed 2 versions, 0.13.6 and 0.13.7 on macOS and it doesn't seem to be sourcing the configuration file located at `~/.config/xplr/init.lua`. I have used a very minimal config where I only modified one keybinding to using a more complex one  that I found online, as well as adding executable permissions to the `init.lua` file. Either way it isn't changed within the TUI of `xplr`.

**Reply by sayanarijit** [reply]
Hi, do you see any log when you press `:l`?

**Reply by sayanarijit** [reply]
Also, posting the `init.lua` will help if possible.

**Reply by lmburns** [reply]
No,  I don't

**Reply by lmburns** [reply]
```lua
version = "0.13.7"

xplr.config.modes.builtin.default.key_bindings.on_key['!'] = {
  help = "shell",
  messages = {
    { Call = { command = "zsh", args = {"-i"} } },
    "ExplorePwdAsync",
    "PopMode",
    "Refresh"
  }
}

-- fzxplr
xplr.config.modes.custom["fzxplr"] = {
  name = "fzxplr",
  key_bindings = {
    on_key = {
      F = {
        help = "search",
        messages = {
          {
            BashExec = [===[
            PTH=$(cat "${XPLR_PIPE_DIRECTORY_NODES_OUT:?}" | awk -F/ '{print $NF}' | fzf)
            if [ -d "$PTH" ]; then
              echo ChangeDirectory: "'"${PWD:?}/${PTH:?}"'" >> "${XPLR_PIPE_MSG_IN:?}"
            elif [ -f "$PTH" ]; then
              echo FocusPath: "'"${PWD:?}/${PTH:?}"'" >> "${XPLR_PIPE_MSG_IN:?}"
            fi
            ]===]
          },
          "PopMode",
        },
      },
    },
    default = {
      messages = {
        "PopMode",
      },
    },
  },
}

xplr.config.modes.builtin.default.key_bindings.on_key["F"] = {
  help = "fzf mode",
  messages = {
    { SwitchModeCustom = "fzxplr" },
  },
}
```

**Reply by sayanarijit** [reply]
This runs fine on my Linux machine.

```
echo $XDG_CONFIG_HOME
```

I'll check Mac meanwhile.

**Reply by lmburns** [reply]
Okay. and yeah I have that variable set:

```sh
 $ echo $XDG_CONFIG_HOME                                                  【 ~/.conf/xplr】─╯
/Users/lucasburns/.config
```
That's why I was wondering if there was a way to see if the configuration file was registering. I tried  `xplr | grep config` and then pressing `#` after it opened and saw matches but didn't see my `init.lua` listed.

**Reply by sayanarijit** [reply]
This should work if you place `init.lua` inside `/Users/lucasburns/.config`.

**Reply by lmburns** [reply]
Like no subfolder `xplr`? Cause I just tried putting it at `$XDG_CONFIG_HOME/init.lua` and `$XDG_CONFIG_HOME/xplr/init.lua` and they both seem to not work. 

Is it supposed tho be listed here anywhere?
```sh
 $ xplr | rg 'config'                                                           【 ~/test】─╯
config:
          config:
                config:
                config:
          config:
                config:
          config:
                config:
          config:
    config:
          config:
          config:
explorer_config:
```

**Reply by sayanarijit** [reply]
Ah right.. my bad.. inside `xplr` subfolder.

**Reply by lmburns** [reply]
I may have found the issue. I'll continue looking here after a little bit, but on macOS, the `dirs::config_dir()` points to `$HOME/Library/Preferences` and ignores the `$XDG_CONFIG_HOME`.  At least if [this](https://docs.rs/dirs/1.0.5/dirs/fn.config_dir.html) is what is used throughout. I don't know any rust so I may be wrong. But if this is the case, it would be nice to have an `XPLR_CONFIG` environment variable. I grepped through the project and didn't find an explicit reference to any `XDG_` variable, so I think its the rust package that is setting that location.

**Reply by sayanarijit** [reply]
Very weird. Thanks for investigating. I'm not sure why my mac works. Will continue investigation later.

**Reply by sayanarijit** [reply]
Moving the thread here https://github.com/sayanarijit/xplr/issues/230
